### PR TITLE
feat(ui): Adjust long texts to better fit on challenge card

### LINKED ISSUE #1948: [Feature] Revisit challenge card designs for those with longer titles
### What projects is this feature for?

OpenChallenges

### Description

Challenge cards with long titles (span over 3+ lines) doesn't look too great, e.g.

Screenshot 2023-08-08 at 10 36 16 PM

We could either:
* increase the card height
* overwrite the line height set for the card-title
* set a character limit for the card title
* hide the platform information if the title is too long

Of these, I like option 2 the best, followed by 1 or 3.  

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by rrchai**
- closes #1948 

## Changelog
- Truncate following challenge's properties that could have long string into two lines with an ellipsis using css approach (`webkit-box`)
  - `name`, `headline`, `incentives`
- Adjust card's width and height
- Adding extra spaces between widgets on the card's footer
- Tested on Chrome, Firefox, Edge ✅ 

>**Note**
>According to the current schemas, the `name` and `headline` of challenge should not have more than 60 and 80 chars respectively . They should be enforced when seeding data via SQL to match the maximum chars defined in schemas. The truncation in this pr is still useful to prevent from this situation and retain the styling of the challenge card.
 
### Before
Screen Shot 2023-08-21 at 7 03 47 PM

### After
Screen Shot 2023-08-21 at 7 02 12 PM

### CODE REVIEW

**Comment by tschaffter** [comment]
I kind of remember a discussion with Verena about 60 char being too short for titles. Even if we end up accepting longer titles, displaying max 60 chars with an ellipse on the card should be a future-proof solution.

**Comment by tschaffter** [comment]
This is kind of a weird case. Maybe we should avoid breaking words with the ellipse?

image

**Comment by rrchai** [comment]
## Updated
If the last word exceeds the maximum limit, remove the entire word as well as special characters.

Screen Shot 2023-08-18 at 6 36 31 PM

**Comment by tschaffter** [comment]
Converting this PR to draft the time to implement the changes discussed this morning.

**Comment by rrchai** [comment]
@tschaffter it seems data response by sorting relevance is returned in the same order now 😮

**Comment by tschaffter** [comment]
@rrchai Are you referring to #1651?

**Comment by rrchai** [comment]
>@rrchai Are you referring to https://github.com/Sage-Bionetworks/sage-monorepo/issues/1651?

Yeah, I tried to refresh multiple times, I got the same order of challenges, at least the first couple lines.

File: libs/openchallenges/styles/src/lib/_general.scss [hunk 1]
```diff
@@ -244,9 +244,16 @@ table {
 }
 
 // CARDS
+@mixin line-camp($n) {
```
Comment (tschaffter) [inline libs/openchallenges/styles/src/lib/_general.scss:247]: Replace `n` by a more descriptive variable.

**Review by tschaffter — APPROVED** [review]
This solution works great! Here are a couple of suggestions:

- rename the variable `n` with a more descriptive name
- update the title and description of this PR since its scope goes beyond the challenge card

**Pull request merged**